openlibrary/plugins/openlibrary/js/add_new_field.js

```diff
@@ -13,90 +13,136 @@
-            $('form').first().add($href).on('submit', function(event) {
-                var array, d, i, data;
-                event.preventDefault();
+        const $href = $(options.href);
+        // handle cancel
+        $href.on('cbox_closed', function() {
 
-                // extract data
-                array = $(this).serializeArray();
-                d = {};
+            if ($selectEl.val() == '__add__') {
+                $selectEl.val('');
+                $selectEl.trigger('focus');
+            }
+            if (options.cancel) {
+                options.cancel();
+            }
+        });
 
-                for (i in array) {
-                    d[array[i].name] = array[i].value.trim();
-                }
+        // handle submit
+        $('form').first().add($href).on('submit', function(event) {
+            event.preventDefault();
 
-                // validate
-                if (options.validate && options.validate.apply($this, [d]) == false) {
-                    return;
-                }
+            // extract data
+            const array = $(event.target).serializeArray();
+            const d = {};
 
-                // close popup
-                $.fn.colorbox.close();
+            for (let i in array) {
+                d[array[i].name] = array[i].value.trim();
+            }
 
-                // add new option
-                $('<option/>')
-                    .html(d.label || d.value)
-                    .attr('value', d.value)
-                    .insertBefore($this.find('option').last().prev()) // insert before ---
-                    .parent().val(d.value);
+            // validate
+            if (options.validate && !options.validate.call($selectEl, d)) {
+                return;
+            }
 
-                // add JSON to hidden field
-                data = null;
-                try {
-                    data = JSON.parse($json.val());
-                }
-                catch (err) {
-                    data = [];
-                }
-                data.push(d);
-                $json.val(JSON.stringify(data));
+            // close popup
+            $.colorbox.close();
+
+            // add new option
+            $('<option/>')
+                .html(d.label || d.value)
+                .attr('value', d.value)
+                .insertBefore($selectEl.find('option').last().prev()) // insert before ---
+                .parent().val(d.value);
 
-                // finally focus the next input field
-                $this.focusNextInputField();
-            });
-            return this;
+            // add JSON to hidden field
+            const data = JSON.parse($json.val());
+            data.push(d);
+            $json.val(JSON.stringify(data));
+
+            // finally focus the next input field
+            $selectEl.focusNextInputField();
         });
-    };
+    }
+
+    function error(error_div, input, message) {
+        $(error_div).show().html(message);
```

Comment:
This isn't displaying the error message divs.  These divs have classes `hidden` and `popalert`, which are both have `display: none`.

Reply:
If the message were displayed, it would probably cause the content div to expand past the modal container:

![Screenshot from 2021-07-12 20-17-08](https://user-images.githubusercontent.com/28732543/125371706-569d4100-e34f-11eb-99bb-f4326c3bfabf.png)

It may be easier to display the error message to the right of the text input.